config/config.go

```diff
@@ -486,6 +495,16 @@ type ExemplarsConfig struct {
 	MaxExemplars int64 `yaml:"max_exemplars,omitempty"`
 }
 
+type TSDBConfig struct {
+	Retention *TSDBRetentionConfig `yaml:"retention,omitempty"`
+}
+
+type TSDBRetentionConfig struct {
+	Time                   model.Duration   `yaml:"time,omitempty"`
+	Size                   units.Base2Bytes `yaml:"size,omitempty"`
+	AllowOverlappingBlocks bool             `yaml:"allow_overlapping_blocks,omitempty"`
```

Comment:
This is not a retention config, so this should come directly under `TSDBConfig`. And it should be a `*bool` so that we don't use it when it's not actually set.

Reply:
When I said "this", I was only pointing to `AllowOverlappingBlocks` :) my bad, I was not clear. Basically the yaml file structure should look like the flag. The dots in the flag will tell you the hierarchy. 
```yaml
storage:
  tsdb:
    allow_overlapping_blocks: true
    retention:
      time: 15d
      size: 10GB
```